data/transactions/logicsig.go

```diff
@@ -25,6 +25,11 @@ import (
 // EvalMaxArgs is the maximum number of arguments to an LSig
 const EvalMaxArgs = 255
 
+// MaxLsigArgSize is the maximum size of an argument to an LSig
+// We use 4096 to match the maximum size of a TEAL value
+// (as defined in `const maxStringSize` in package logic)
+const MaxLogicSigArgSize = 4096
```

Comment:
Should we export that constant and use it directly?

Reply:
It would be nice, but `package logic` already imports `package transactions` and we cannot have circular imports.
We would need to define the constant in `package transactions` and then import it in `package logic`.

Or alternatively make it a consensus parameter; in this case I guess also [const EvalMaxArgs = 255](https://github.com/algorand/go-algorand/pull/6057/files/4de5dd0d3470ac252cc35343438c7ddb21c805ea#diff-8f4a7b812c0892353812d410dcea94ccef1b18277f6c33b4e56437d27c39b134R26) could be made a consensus parameter

Reply:
I'm ok leaving it as is.